duckduckgo/Android
# Move FaviconManager interface to BrowserApi module

Task/Issue URL: https://app.asana.com/0/488551667048375/1202646807324435/f

### Description
Move `FaviconManager` out of `common-ui` and into `browser-api` module

### Steps to test this PR

- Add a bookmark, then visit the bookmarks screen. Ensure favicon shows and nothing crashes.
